app/services/update_account_service.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class UpdateAccountService
+  def initialize(account, params)
+    @account = account
+    @params = params
+    @account_share = @account.account_shares.where.not(parental_key: nil).first
+
+    return unless @account_share
```

Comment:
WDYT if account_share and all these entities under the condition will be private memoized methods? Would it make the code cleaner and easier to read?